astronomer/providers/sftp/example_dags/example_sftp.py

```diff
@@ -16,6 +46,168 @@
+        MaxCount=1,
+        InstanceType=SFTP_INSTANCE_TYPE,
+        KeyName=PEM_FILENAME,
+        SecurityGroups=[INBOUND_SECURITY_GROUP],
+    )
+    instance_id = instance[0].id
+    ti = get_current_context()["ti"]
+    ti.xcom_push(key="ec2_instance_id", value=instance_id)
+
+
+def get_ec2_instance_details(task_instance: "TaskInstance") -> None:
+    """Get the EC2 instance details by id retrieved from the Xcom"""
+    import boto3
+
+    client = boto3.client("ec2", **AWS_S3_CREDS)
+    ec2_instance_id_xcom = task_instance.xcom_pull(key="ec2_instance_id", task_ids=["create_ec2_instance"])[0]
+    response = client.describe_instances(
+        InstanceIds=[ec2_instance_id_xcom],
+    )
+    instance_details = response["Reservations"][0]["Instances"][0]
+    ti = get_current_context()["ti"]
+    ti.xcom_push(
+        key="instance_response_master_security_group", value=instance_details["SecurityGroups"][0]["GroupId"]
+    )
+    ti.xcom_push(key="instance_public_dns_name", value=instance_details["PublicDnsName"])
+
+
+def add_inbound_rule_for_security_group(task_instance: "TaskInstance") -> None:
+    """
+    Sets the inbound rule for the aws security group, based on
+    current ip address of the system.
+    """
+    import boto3
+    from botocore.exceptions import ClientError
+
+    client = boto3.client("ec2", **AWS_S3_CREDS)
+    current_docker_ip = get("https://api.ipify.org").text
+
+    # Allow SSH traffic on port 22 and copy file to ec2 instance.
+    try:
+        client.authorize_security_group_ingress(
+            GroupId=task_instance.xcom_pull(
+                key="instance_response_master_security_group", task_ids=["get_instance_details"]
+            )[0],
+            IpPermissions=[
+                {
+                    "IpProtocol": "tcp",
+                    "FromPort": 22,
+                    "ToPort": 22,
+                    "IpRanges": [{"CidrIp": str(current_docker_ip) + "/32"}],
+                }
+            ],
+        )
+    except ClientError as error:
+        if error.response.get("Error", {}).get("Code", "") == BOTO_DUPLICATE_PERMISSION_ERROR:
+            logging.error(
+                "Ingress for port 22 already authorized. Error message is: %s",
+                error.response["Error"]["Message"],
+            )
+        else:
+            raise error
+
+
+def ssh_and_run_command(task_instance: Any, **kwargs: Any) -> None:
+    """
+    Load the private_key from airflow variable and creates a pem_file
+    at /tmp/. SSH into the machine and execute the bash script from the list
+    of commands.
+    """
+    # remove the file if it exists
+    if os.path.exists(f"/tmp/{PEM_FILENAME}.pem"):
+        os.remove(f"/tmp/{PEM_FILENAME}.pem")
+
+    # read the content for pem file from Variable set on Airflow UI.
+    with open(f"/tmp/{PEM_FILENAME}.pem", "w+") as fh:
+        fh.write(PRIVATE_KEY)
+
+    shutil.copyfile(f"/tmp/{PEM_FILENAME}.pem", f"/usr/local/airflow/dags/{PEM_FILENAME}.pem")
```

Comment:
Why do we need to copy this file to the dags folder?

Reply:
I have copied this file to dags folder because it will in a shared location, so that trigger also will have access to the file

Reply:
Currently, this async version doesn't support private key authentication used to connect to the SFTP server, so we need to add support for that created [ticket](https://app.zenhub.com/workspaces/airflow-providers-team-61a89ee7fbf048001dc8378d/issues/astronomer/astronomer-providers/760). Once it is implemented we can remove this copying file to the dag folder we can add private_key option in the connection

Reply:
sounds good to me @bharanidharan14

Reply:
I would say we can fix the [issue](https://app.zenhub.com/workspaces/airflow-providers-team-61a89ee7fbf048001dc8378d/issues/astronomer/astronomer-providers/760) and then merge it since we would be seeing intermittent failures till then on the Astro cloud rather than that we can fix properly and then merge it.

Reply:
Agree with @rajaths010494

Reply:
@rajaths010494 Added support for private key and add test case for that  and removed copying of file to dag folder is removed

Reply:
@pankajkoti can you review this now I have addressed adding the private key